src/arch/arm/kernel/boot.c

```diff
@@ -305,6 +313,18 @@ BOOT_CODE static void release_secondary_cpus(void)
     while (ksNumCPUs != CONFIG_MAX_NUM_NODES) {
         /* perform a memory release+acquire to get new values of ksNumCPUs */
         __atomic_signal_fence(__ATOMIC_ACQ_REL);
+#if defined(CONFIG_KERNEL_MCS) && defined(CONFIG_DEBUG_BUILD)
+        /* Test whether clocks are synchronised across nodes: */
+        ticks_t t = getCurrentTime();
+        for (int i = 1; i < CONFIG_MAX_NUM_NODES; ++i) {
+            if (clock_sync_test[i] != 0) {
+                ticks_t clock_diff = t - clock_sync_test[i];
+
+                assert(clock_diff >= 0 && clock_diff <= getTimerPrecision());
```

Comment:
Does this wok, with `ticks_t` an aliad of `uint64_t` this can't be negative? Maybe use:
```suggestion
                assert((t >=clock_sync_test[i]) && (t - clock_sync_test[i] <= getTimerPrecision()));
```